mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -269,6 +270,8 @@ class LDA private (
    */
   @Since("1.3.0")
   def setCheckpointInterval(checkpointInterval: Int): this.type = {
+    require(checkpointInterval == -1 || checkpointInterval > 0,
```

Comment:
It has been documented at [here](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/impl/PeriodicGraphCheckpointer.scala#L73), but I think it's better to also clarify at ```setCheckpointInterval```.